src/expressions.rst

```diff
@@ -2704,78 +2699,77 @@ Array and Slice Indexing Expressions
 .. rubric:: Legality Rules
 
 :dp:`fls_42ijvuqqqlvh`
-An :t:`array index expression` is an :t:`expression` that indexes into an
-:t:`array` or a :t:`slice`.
+An :t:`index expression` is an :t:`expression` that indexes into a :t:`value`
+of a :t:`type`.
```

Comment:
I wonder whether we should introduce the concept of an "indexable type", something along the lines of "An indexable type is a type that implements the core::ops::Index trait.". I don't think that the phrasing "... indexes into a value of a type" is entirely correct, as theoretically I can index into an integer.

Reply:
> I don't think that the phrasing "... indexes into a value of a type" is entirely correct, as theoretically I can index into an integer.

Why is that not correct? A value of a type also applies to an integer
And yes I agree, a new term for indexable type might be good to add

Reply:
I was arguing that "type" is not accurate enough, it should be "indexable type" as you can't index into an arbitrary type.